metals/src/main/scala/scala/meta/internal/metals/Embedded.scala

```diff
@@ -40,23 +60,34 @@ final class Embedded(
   ): PresentationCompiler = {
     val classloader = presentationCompilers.getOrElseUpdate(
       ScalaVersions.dropVendorSuffix(info.getScalaVersion),
-      statusBar.trackSlowTask("Downloading presentation compiler") {
+      statusBar.trackSlowTask("Preparing presentation compiler") {
         Embedded.newPresentationCompilerClassLoader(info, scalac)
       }
     )
-    val services =
-      ServiceLoader.load(classOf[PresentationCompiler], classloader).iterator()
+    serviceLoader(
+      classOf[PresentationCompiler],
+      classOf[ScalaPresentationCompiler].getName(),
+      classloader
+    )
+  }
+
+  private def serviceLoader[T](
+      cls: Class[T],
+      className: String,
+      classloader: URLClassLoader
+  ): T = {
+    val services = ServiceLoader.load(cls, classloader).iterator()
     if (services.hasNext) services.next()
     else {
-      // NOTE(olafur): ServiceLoader doesn't find the presentation compiler service
-      // on Appveyor for some reason, I'm unable to reproduce on my computer. Here below
-      // we fallback to manual classloading.
-      val cls =
-        classloader.loadClass(classOf[ScalaPresentationCompiler].getName)
+      // NOTE(olafur): ServiceLoader doesn't find the service on Appveyor for
```

Comment:
We no longer use AppVeyor, do we need to load ScalaPresentationCompiler explicitely?

Reply:
This is a reasonable fallback mechanism if the service loader doesn't work for some reason. This code was here before so I don't stir up things in this PR